polkadot/primitives/src/v6/executor_params.rs

```diff
@@ -130,6 +182,126 @@ impl ExecutorParams {
 		}
 		None
 	}
+
+	/// Check params coherence.
+	pub fn check_consistency(&self) -> Result<(), ExecutorParamError> {
+		use ExecutorParam::*;
+		use ExecutorParamError::*;
+
+		let mut seen = BTreeMap::<&str, u64>::new();
+
+		macro_rules! check {
```

Comment:
I'm loving the checks you added, but is there a reason we use a macro instead of a simple fn here?

Reply:
The function implementation might be more complicated: more inputs, branches, and more jumps. Rough idea:

```
fn check_param(seen, name, val, predicate) -> Result
check(seen, name, val, predicate)?
```

Using macro here seems simpler.